src/CanvasRenderingContext2d.cc

```diff
@@ -292,6 +292,65 @@ Context2d::restorePath() {
   cairo_path_destroy(_path);
 }
 
+/*
+ * Create temporary surface for gradient or pattern transparency
+ */
+cairo_pattern_t*
+create_transparent_gradient(cairo_pattern_t *source, float alpha) {
+  double x0;
+  double y0;
+  double x1;
+  double y1;
+  double r0;
+  double r1;
+  int count;
+  int i;
+  double offset;
+  double r;
+  double g;
+  double b;
+  double a;
+  cairo_pattern_t *newGradient;
+  cairo_pattern_type_t type = cairo_pattern_get_type(source);
+  cairo_pattern_get_color_stop_count(source, &count);
+  if (type == CAIRO_PATTERN_TYPE_LINEAR) {
+    cairo_pattern_get_linear_points (source, &x0, &y0, &x1, &y1);
+    newGradient = cairo_pattern_create_linear(x0, y0, x1, y1);
+  } else if (type == CAIRO_PATTERN_TYPE_RADIAL) {
+    cairo_pattern_get_radial_circles(source, &x0, &y0, &r0, &x1, &y1, &r1);
+    newGradient = cairo_pattern_create_radial(x0, y0, r0, x1, y1, r1);
+  } else {
+    Nan::ThrowError("Unexpected gradient type");
```

Comment:
I don't remember exactly how this works but it feels like there should be a `return` statement here:

```c++
Nan::ThrowError("Unexpected gradient type");
return nullptr;
```

Hmm, actually, possibly that needs to be guarded at the call-site as well? I don't see how `Nan::ThrowError` can alter the normal flow of a c++ function, but maybe I'm missing something 🤔

Reply:
if i only knew c++... i have C knowledge coming from embedded platforms with proprietary api. I stop there.

How can i test it? i can force a gradient type comparision and see what happens?

Reply:
> i can force a gradient type comparision and see what happens?

that sounds good 👍

Reply:
oh by the way using nullptr kills nodejs < 4.